basic_bootloader/src/bootloader/transaction/mod.rs

```diff
@@ -1,1467 +1,322 @@
-                }
-            }
-        }
-
-        // paymasters are not supported, even for EIP712 txs
-        if !self.paymaster_input.range.is_empty() {
-            return Err(());
-        }
-
-        // factory deps allowed only for eip712, or l1 to l2/upgrade txs
-        // we ignore factory deps, as deployments performed via bytecode,
-        // but we allowed them for backward compatibility with some Era VM tests
-        match tx_type {
-            Self::EIP_712_TX_TYPE | Self::L1_L2_TX_TYPE | Self::UPGRADE_TX_TYPE => {}
-            _ => {
-                if !self.factory_deps.range.is_empty() {
-                    return Err(());
-                }
-            }
-        }
-
-        // Access list is empty except for txs that support it
-        match tx_type {
-            Self::EIP_1559_TX_TYPE | Self::EIP_2930_TX_TYPE => {}
-            #[cfg(feature = "pectra")]
-            Self::EIP_7702_TX_TYPE => {}
-            _ => {
-                if !self
-                    .reserved_dynamic
-                    .access_list_is_empty(&self.underlying_buffer)?
-                {
-                    return Err(());
-                }
+impl<A: Allocator> Transaction<A> {
+    /// Parse a transaction from a raw buffer using the system IO oracle.
+    pub fn try_from_buffer<
+        S: EthereumLikeTypes<
+            Metadata = zk_ee::system::metadata::zk_metadata::ZkMetadata,
+            Allocator = A,
+        >,
+    >(
+        buffer: UsizeAlignedByteBox<A>,
+        system: &mut System<S>,
+    ) -> Result<Self, TxError>
+    where
+        S::IO: IOSubsystemExt,
+    {
+        use zk_ee::oracle::IOOracle;
+        let expected_chain_id = system.get_chain_id();
+
+        // query the transaction encoding format from the oracle
+        let format: TxEncodingFormat = match system
+            .io
+            .oracle()
+            .query_with_empty_input(TX_ENCODING_FORMAT_QUERY_ID)
+        {
+            Ok(format) => format,
+            Err(e) => {
+                return Err(e.into());
             }
-        }
+        };
 
-        // Authorization list is empty except for txs that support it
-        // For EIP 7702, authorization list cannot be empty
-        #[cfg(feature = "pectra")]
-        match tx_type {
-            Self::EIP_7702_TX_TYPE => {
-                if self
-                    .reserved_dynamic
-                    .authorization_list_is_empty(&self.underlying_buffer)?
-                {
-                    return Err(());
-                }
-            }
-            _ => {
-                if !self
-                    .reserved_dynamic
-                    .authorization_list_is_empty(&self.underlying_buffer)?
-                {
-                    return Err(());
-                }
+        match format {
+            TxEncodingFormat::Eth => {
+                // RLP-encoded transactions don't include the `from` field, so we need to query it from the oracle.
```

Comment:
Note: they do not include it because `from` can be derived. We also can derive them, but probably the current approach is simpler: we skip ecrecover for simulations